### DISCUSSION TITLE
manualPagination does not work when useGroupBy is included

### DISCUSSION THREAD

**Discussion by johnathanludwig** [TanStack/table#2741] [Ideas]
**Describe the bug**
`manualPagination` is broken when `useGroupBy` is included. When going to the next page, `useGroupBy` causes the page to be set back to the previous page. 

**Provide an example via Codesandbox!**
This is the same `manualPagination` sandbox from the documentation, but with `useGroupBy` included on line 78.

https://codesandbox.io/s/manualpagination-bug-with-usegroupby-ph2jq?file=/src/App.js

**Steps To Reproduce**

1. Click the next page arrow button.
2. The page goes to the next page for a fraction of a second before going back

**Expected behavior**
The table should stay on the next page.

**Desktop:**

- OS: MacOS 10.15.6
- Browser Firefox
- Version 80

**Additional context**
`useGroupBy` does not need any extra configuration, just being in the plugins along with `usePagination` causes the bug. `disableGroupBy` does not prevent the bug, however `manualGroupBy` does.

**Comment by tannerlinsley** [comment]
Thanks for noting this issue. To fix it, a breaking change is required and thus we will slate the fix for v8 and, for sake of triage, close this issue.

Thanks for your patience!
